--- ../alsa-kernel/firewire/cmp.c	2012-05-31 21:22:53.000000000 +0200
+++ cmp.c	2012-06-01 08:39:42.000000000 +0200
@@ -1,3 +1,11 @@
+#define __NO_VERSION__
+#include "adriver.h"
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 5, 0)
+const char *fw_rcode_string(int rcode);
+#endif
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 39)
+#include "iso-resources-old.h"
+#endif
 /*
  * Connection Management Procedures (IEC 61883-1) helper functions
  *
